src/main/java/uk/ac/cam/cl/dtg/segue/api/GlossaryFacade.java

```diff
@@ -0,0 +1,167 @@
+import com.google.inject.Inject;
+import com.google.inject.name.Named;
+import io.swagger.annotations.Api;
+import io.swagger.annotations.ApiOperation;
+import org.apache.http.HttpResponse;
+import org.apache.http.client.HttpClient;
+import org.apache.http.client.methods.HttpGet;
+import org.apache.http.impl.client.DefaultHttpClient;
+import org.jboss.resteasy.annotations.GZIP;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import uk.ac.cam.cl.dtg.segue.configuration.SegueGuiceConfigurationModule;
+import uk.ac.cam.cl.dtg.segue.dao.ILogManager;
+import uk.ac.cam.cl.dtg.segue.dao.content.ContentManagerException;
+import uk.ac.cam.cl.dtg.segue.dao.content.IContentManager;
+import uk.ac.cam.cl.dtg.segue.dos.content.Content;
+import uk.ac.cam.cl.dtg.segue.dto.ResultsWrapper;
+import uk.ac.cam.cl.dtg.segue.dto.SegueErrorResponse;
+import uk.ac.cam.cl.dtg.segue.dto.content.ContentDTO;
+import uk.ac.cam.cl.dtg.util.PropertiesLoader;
+
+import javax.ws.rs.*;
+import javax.ws.rs.core.Context;
+import javax.ws.rs.core.EntityTag;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Request;
+import javax.ws.rs.core.Response;
+import javax.ws.rs.core.Response.Status;
+import javax.xml.bind.annotation.XmlElement;
+import java.io.IOException;
+import java.util.*;
+
+import static com.google.common.collect.Maps.immutableEntry;
+import static uk.ac.cam.cl.dtg.segue.api.Constants.*;
+
+/**
+ * Glossary Facade
+ *
+ * This facade is intended to provide access to glossary terms.
+ *
+ */
+@Path("/glossary")
+@Api(value = "/glossary")
+public class GlossaryFacade extends AbstractSegueFacade {
+    private static final Logger log = LoggerFactory.getLogger(GlossaryFacade.class);
+
+    private final IContentManager contentManager;
+    private final String contentIndex;
+
+    /**
+     * @param properties     - to allow access to system properties.
+     * @param contentManager - so that metadata about content can be accessed.
+     * @param contentIndex   - to access the right version of the content.
+     * @param logManager     - for logging events using the logging api.
+     */
+    @Inject
+    public GlossaryFacade(final PropertiesLoader properties, final IContentManager contentManager,
+                          @Named(CONTENT_INDEX) final String contentIndex,
+                          final ILogManager logManager) {
+        super(properties, logManager);
+        this.contentManager = contentManager;
+        this.contentIndex = contentIndex;
+    }
+
+    /**
+     * Gets all the glossary terms that are indexed.
+     *
+     * @param limit      - Maximum amount of terms to retrieve. Used for pagination.
+     * @param startIndex - Index from which to start retrieving when results exceed limit.
+     *
+     * @return Paginated list of glossary terms.
+     */
+    @GET
+    @Path("terms")
+    @Produces(MediaType.APPLICATION_JSON)
+    @ApiOperation(value = "Get all the glossary terms that are indexed.")
```

Comment:
If we have more terms than the DEFAULT_RESULTS_LIMIT this will not return all of the glossary terms. Could lead to some hard to find bugs. Do we need the start_index and limit for this query?

Reply:
The object returned is something like this:

```
{
  "results": [
    // ...
  "totalResults": 125
}
```

this is for any limit that is passed with the query, so if you get `DEFAULT_RESULTS_LIMIT` (which is 10, I think) you should also check `totalResults` to see if there are more available, and then it's your responsibility to query for more, using the `start_index` parameter.

Currently, I hard-coded 10000 in the `isaac-cs-app`, which I believe is well beyond what we could even get as far as glossary terms go, but I agree that this can lead to weird bugs.